docs-2.0/4.deployment-and-installation/3.upgrade-nebula-graph.md

```diff
@@ -0,0 +1,141 @@
+# 升级Nebula Graph版本
+
+Nebula Graph暂不支持一键升级版本，本文介绍如何手动升级版本。
+
+## 升级限制
+
+- 当前仅支持如下版本的升级，其他版本的升级未验证。
+
+  - Nebula Graph 1.2升级至Nebula Graph 2.0.0-GA
+
+  - Nebula Graph 2.0.0-RC1升级至Nebula Graph 2.0.0-GA
+
+- 必须在原机器进行升级，不能修改机器的IP地址，以及配置文件的IP地址和端口。
+
+- 硬盘空间要求：硬盘剩余空间≥三倍Nebula Graph data目录空间。
+
+- 升级时需要停止Nebula Graph，具体升级所需时间和机器配置、数据规模、并发参数有关。
+
+- 升级完成后，集群对外服务端口会有变化，请查看对应服务的配置文件。
+
+## 升级时间
+
+本文测试升级的环境如下：
+
+- 机器配置：32核CPU、62 GB内存、SSD
+
+- 数据规模：Nebula Graph1.2版本LDBC测试数据100 GB（1个图空间、24个分片、data目录92 GB）
+
+- 并发参数：`--max_concurrent=5`、`--max_concurrent_parts=24`、`--write_batch_num=100`
+
+升级共耗时21分钟（压缩耗时13分钟）。
+
+您可以根据业务情况调整并发参数，并发参数说明如下。
+
+|参数名称|默认值|
+|:---|:---|
+|`--max_concurrent`|5|
+|`--max_concurrent_parts`|10|
+|`--write_batch_num`|100|
+
+## 路径说明
+
+不同版本的Nebula Graph安装路径不同，请使用实际路径。本文示例为：
+
+- 旧版本（Nebula Graph 1.2/2.0.0-RC1）：`/usr/local/nebula/`
+
+- 新版本（Nebula Graph 2.0.0-GA）：`/usr/local/nebula_GA/`
+
+## 升级步骤
+
+1. 安装新版本Nebula Graph，保证配置和旧版本一致。
+
+    >**注意**：外围服务都需要修改为新版本端口，例如Nebula Exchange、Nebula Importer、Nebula Console等。
+
+2. 停止旧版本的Nebula Graph服务。
+
+    ```bash
+    $ sudo usr/local/nebula/scripts/nebula.service stop all
+    ```
+
+3. 将旧版本的目录`data/meta`复制到新版本的相同位置。
+
+    >**说明**：目录`data/meta`由配置文件`nebula-metad.conf`中的参数`--data_path`指定。
+
+    ```bash
+    $ sudo cp -R /usr/local/nebula/data/meta/ /usr/local/nebula_GA/data/meta/
+    ```
+
+4. 启动新版本的Meta服务。集群部署时，需要将集群中的所有Meta服务都启动。
+
+    >**说明**：启动前请确认对如下两项配置是否有需求，如果需要，请修改配置文件`nebula-metad.conf`。
+    >
+    >- `--null_type`：升级后的schema字段是否要支持`NULL`，默认为支持。如果不希望支持，请设置为`false`，但是后续添加字段时需要设置默认值，否则无法读取数据。
+    >- `--string_index_limit`：设置升级后旧版里的string类型字段的的索引长度，默认为64。
+
+    ```bash
+    $ sudo /usr/local/nebula_GA/scripts/nebula.service start metad
+    ```
+
+5. 检查Meta服务是否正常。
+
+    ```bash
+    $ sudo /usr/local/nebula_GA/scripts/nebula.service status metad
+    ```
+
+    如果服务异常，请查看目录`/usr/local/nebula_GA/logs`内的meta相关日志排查故障。
+
+6. 执行如下命令升级版本。集群部署时，需要在集群中的所有Storage服务节点都执行此命令。
+
+    如果需要缩短升级时间，可以在以下命令中增加[并发相关参数](#升级时间)。
+
+    ```bash
+    $ sudo /usr/local/nebula_GA/bin/db_upgrader  \
+    --src_db_path=<old_storage_directory_path> \
+    --dst_db_path=<new_storage_directory_path>  \
+    --upgrade_meta_server=<meta_server_ip1>:<meta_server_port1>[,<meta_server_ip2>:<meta_server_port2>..] \
+    --upgrade_version=<old_nebula_version> \
+    ```
+
+    示例如下：
+
+    ```bash
+    $ sudo /usr/local/nebula_GA/bin/db_upgrader \
+    --src_db_path=/usr/local/nebula/data/storage/ \
```

Comment:
Add some description about where data/storage is from. Add a example when data_path consists of multiple path if necessary.